### ISSUE TITLE
Fuel: tests failing on WIN

### ISSUE THREAD

**Issue by MarcusDenker** [guillep/pharo-core#9305]
**Bug description**
Since the update of Fuel, we have these tests failing (on windows only)

```
windows-64 / Tests-windows-64 / testDate – Windows64.Fuel.Tests.Core.Base.FLBasicSerializationTest
windows-64 / Tests-windows-64 / testDate – Windows64.Fuel.Tests.Core.Base.FLBinaryFileStreamBasicSerializationTest
windows-64 / Tests-windows-64 / testDate – Windows64.Fuel.Tests.Core.Base.FLByteArrayBasicSerializationTest
windows-64 / Tests-windows-64 / testDate – Windows64.Fuel.Tests.Core.Base.FLFileReferenceStreamBasicSerializationTest
windows-64 / Tests-windows-64 / testDate – Windows64.Fuel.Tests.Core.Base.FLGZippedBasicSerializationTest
windows-64 / Tests-windows-64 / testDate – Windows64.Fuel.Tests.Core.Base.FLInMemoryBasicSerializationTest
windows-64 / Tests-windows-64 / testDate – Windows64.Fuel.Tests.Core.Base.FLProcessBasicSerializationTest
```

**Comment by theseion** [comment]
That's weird. My tests are green: https://ci.appveyor.com/project/theseion/fuel/build/job/r6blvp47vxgj5csw

<blockquote><img src="/assets/images/favicon-blue.ico" width="48" align="right"><div><strong><a href="https://ci.appveyor.com/project/theseion/fuel/build/job/r6blvp47vxgj5csw">AppVeyor</a></strong></div></blockquote>

**Comment by theseion** [comment]
Actually, I remember seeing those. But they disappeared when I pushed another commit, so I didn't think anything of it...

**Comment by theseion** [comment]
All of them are `#testDate`. I will boot up a Windows VM and try to see what's happening.

**Comment by theseion** [comment]
It's a bug in the refactored integer encoding logic in Fuel (negative integers specifically). I'll fix that and open a PR ASAP.

The reason the bug showed up is probably because of the time zone setting on the Windows machine. Luckily, I have a time zone of -7 in my Windows VM, otherwise I wouldn't have caught it.

**Comment by theseion** [comment]
I've made the fix and opened an [PR](#9334). The PR contains a couple of extra things that I've updated for 3.0.3 but weren't important enough to integrate.
